Nodejs/Product/Nodejs/SharedProject/ProjectNode.cs

```diff
@@ -6026,57 +6028,44 @@ private void SetupProjectGlobalPropertiesThatAllProjectSystemsMustSet()
         /// </remarks>
         private bool TryBeginBuild(bool designTime, bool requiresUIThread = false)
         {
-            var releaseUIThread = false;
-
-            try
+            // If the SVsBuildManagerAccessor service is absent, we're not running within Visual Studio.
+            if (this.buildManagerAccessor != null)
             {
-                // If the SVsBuildManagerAccessor service is absent, we're not running within Visual Studio.
-                if (this.buildManagerAccessor != null)
+                if (requiresUIThread)
                 {
-                    if (requiresUIThread)
+                    var result = this.buildManagerAccessor.ClaimUIThreadForBuild();
```

Comment:
Can `ClaimUIThreadForBuild` lock the UI thread and then throw and exception?

Reply:
I'm not sure what you mean by "lock the UI thread".  It sets a flag indicating that it's going to use the UI thread and that others checking the flag should not.

I would also not expect it to throw since it returns an HRESULT.  It can still throw OOM, etc, but such exceptions are unrecoverable anyway.